crates/swc_ecma_transforms_react/src/refresh/mod.rs

```diff
@@ -869,12 +383,20 @@ impl<C: Comments> Fold for Refresh<C> {
                 _ => Persist::None,
             };
 
-            items.push(item);
+            if let Persist::Hoc(_) = persistent_id {
+                // we need to make hook transform happens after component for
+                // HOC
+                items.push(item);
+            } else {
+                item.visit_mut_children_with(&mut hook_visitor);
+                self.scope_binding.truncate(current_binding);
+
+                items.push(item);
+                let stmts = hook_visitor.extra_stmt.take();
```

Comment:
`items.extend(hook_visitor.extra_stmt.take().into_iter().map(ModuleItem::Stmt))` would be better for performance because of `TrustedLen`.